source/CMakeLists.txt

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.13)
+
+set(CMAKE_C_STANDARD 99)
+
+# Azure IoT FreeRTOS Middleware Library
+add_library(az_iot_middleware_freertos
+  ${CMAKE_CURRENT_LIST_DIR}/azure_iot_hub_client.c
+  ${CMAKE_CURRENT_LIST_DIR}/azure_iot_provisioning_client.c
+  ${CMAKE_CURRENT_LIST_DIR}/azure_iot.c
+)
+
+# The user needs to supply this
+target_link_libraries(az_iot_middleware_freertos
+  PUBLIC
+    middleware_config)
+
+target_link_libraries(az_iot_middleware_freertos
+  PUBLIC
+    az::iot::common
+    az::iot::hub
+    az::iot::provisioning
+)
+
+# Include port files
+# TODO: Have the user select the port path to use
+if(WIN32)
+  target_include_directories(az_iot_middleware_freertos
+    PRIVATE
+      ${az_iot_middleware_freertos_SOURCE_DIR}/libraries/FreeRTOS/FreeRTOS/Source/portable/MSVC-MingW
```

Comment:
Can we have port target of FreeRTOS and user define it ? Here we just consume it

Reply:
This part requires a bit of thinking I feel. We could go the target route which requires a bit of tweaking on their end since they might not have it. They would have to tweak their own CMake build to construct something that WE need. Fundamentally we just need a path so maybe we ask them to set an environment variable kind of like the embedded for the `VCPKG_ROOT`? That way less of our build requirements leak into their build process?